### PR TITLE
Use `get_hostname` instead of `socket.getfqdn`

### PR DESCRIPTION
**Pull request by jedcunningham**
We allow users to configure a different function to determine the
hostname, so we should use that consistently when we need the hostname.

### CODE REVIEW

**Review by kaxil — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Wondering . Would that be a reason why there are entries with "nul" hostname ? I was asking the question at the development channel in slack as we seem to have repeating case where "log cannot be retrieved as hostname is missing".

**Comment by jedcunningham** [comment]
The only thing I can think of is if the worker isn't actually able to set the hostname on the TI, so it falls back to trying to get the log from the worker with "null" in that column? The [log hostname side of things](https://github.com/apache/airflow/blob/42abbf0d61f94ec50026af0c0f95eb378e403042/airflow/models/taskinstance.py#L1264) already used the helper, so I don't think this would help.

File: airflow/www/auth.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def decorated(*args, **kwargs):
                 return (
                     render_template(
                         'airflow/no_roles_permissions.html',
-                        hostname=socket.getfqdn()
+                        hostname=get_hostname
```
Comment (mik-laj) [inline airflow/www/auth.py]:
```suggestion
                        hostname=get_hostname()
```

Comment (jedcunningham) [reply airflow/www/auth.py]: Oops, thanks!

**Review by dstandish — APPROVED** [review]

**Review by uranusjr — APPROVED** [review]

**Review by mik-laj — APPROVED** [review]

File: airflow/www/auth.py (hunk 1 above)
Comment (uranusjr) [reply airflow/www/auth.py]: This one seems to have got lost during the rebase

**Comment by uranusjr** [comment]
Looks like GitHub is having some issues.

**Comment by potiuk** [comment]
Reopened - github seems to be back up

**Pull request merged**